internal/hammer/loadtest/workers.go

```diff
@@ -208,18 +229,41 @@ func (w *LogWriter) Run(ctx context.Context) {
 		}
 		newLeaf := w.gen()
 		lt := LeafTime{QueuedAt: time.Now()}
-		index, err := w.writer(ctx, newLeaf)
+		index, timestamp, err := w.writer(ctx, newLeaf)
 		if err != nil {
 			w.errChan <- fmt.Errorf("failed to create request: %w", err)
 			continue
 		}
 		lt.Index, lt.AssignedAt = index, time.Now()
-		// See if we can send a leaf sample
+
+		// See if we can send a leaf sample.
 		select {
 		// TODO: we might want to count dropped samples, and/or make sampling a bit more statistical.
 		case w.leafChan <- lt:
 		default:
 		}
+
+		// TODO: Remove the json.Unmarshal by generating the chain and
+		// marshaling the add chain request from w.gen() at a later stage.
+		var req types.AddChainRequest
+		if err := json.Unmarshal(newLeaf, &req); err != nil {
+			klog.Warningf("Failed to unmarshal add-chain request: %v", err)
+		}
+		var chain []byte
+		for _, cert := range req.Chain {
+			chain = append(chain, cert...)
+		}
+
+		// Send LeafMMD for inclusion proof verification.
+		select {
+		case w.leafMMDChan <- LeafMMD{chain, index, timestamp}:
+		default:
+			// Drop if leafMMDChan is full. This could happen if the MMD verifiers are falling behind.
+			if len(w.leafMMDChan) > 0 {
```

Comment:
```suggestion
			if cap(w.leafMMDChan) > 0 {
```

The length of a channel can fluctuate. I think you're after the assigned capacity?

Reply:
Good catch! Updated.